### PR TITLE
py-chainer: Add test method for ChainerMN (continued #21848, #21940)

### PR DESCRIPTION
**Pull request by a-saitoh-fj**
continued #21848, #21940
Add test method for ChainerMN .
Removed the environment variable settings in the Fugaku file pointed out in #21940.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-chainer/package.py [hunk 1]
```diff
@@ -34,3 +38,49 @@ class PyChainer(PythonPackage):
...
+                options=opts,
+                work_dir=test_dir,
+            )
+
+            # check results
+            json_open = open(join_path(test_dir, 'log'), 'r')
+            json_load = json.load(json_open)
+            v = {d.get('epoch'): d.get('main/accuracy') for d in json_load}
+            if 1 not in v or 20 not in v:
+                raise RuntimeError('Cannot found epoch 1 or epoch 20')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-chainer/package.py]:
```suggestion
                raise RuntimeError('Cannot find epoch 1 or epoch 20')
```

**Comment by a-saitoh-fj** [comment]
py-chainer: Fixed the word in the message
found -> find

**Comment by a-saitoh-fj** [comment]
I am sorry. Not enough fixes.

Deleted unnecessary imports.
Please check this recipe.

**Review by adamjstewart — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-chainer/package.py [hunk 2]
```diff
@@ -34,3 +37,49 @@ class PyChainer(PythonPackage):
...
+            self.run_test(
+                mpi_name,
+                options=opts,
+                work_dir=test_dir,
+            )
+
+            # check results
+            json_open = open(join_path(test_dir, 'log'), 'r')
+            json_load = json.load(json_open)
+            v = {d.get('epoch'): d.get('main/accuracy') for d in json_load}
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-chainer/package.py]: Dict comprehension doesn't work in old versions of Python, you'll have to use a for-loop.

**Comment by a-saitoh-fj** [comment]
@adamjstewart
I'm sorry
Incorporation of the measures pointed out in #21940 was insufficient. 
(@alalazo pointed out, and showed me how to fix it.)

Check out this recipe, please

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**